### ISSUE TITLE
Make Event_System optional so gateway can build without Conditional library

### ISSUE THREAD

**Issue by darobs** [Azure/iot-edge#232]
dependencies.cmake:

```c
findAndInstall(azure_c_shared_utility 1.0.25
    ${PROJECT_SOURCE_DIR}/deps/c-utility
    ${PROJECT_SOURCE_DIR}/deps/c-utility
    -Duse_installed_dependencies=ON
    -Drun_unittests=${run_unittests}
    -Dbuild_as_dynamic=ON
    -Duse_default_uuid=${use_xplat_uuid}
    ${PASSVARS}
    -G "${CMAKE_GENERATOR}")
```

There are more options in the root CMakeLists.txt that should be passed into this call, specifically "use_condition".

It would be even better if there was a more generic way to pass options down from top-level CmakeLists.txt to submodules.

**Comment by damonbarry** [comment]
@darobs The gateway needs the Condition API for the experimental Events stuff. So instead of introducing `use_condition` in our CMakeLists.txt, we should introduce a CMake option (OFF by default) to enable the Event system feature. If that flag is ON then we should:
1. Set `use_condition=ON` when we build c-utility, and
2. Build the event system stuff in our code

...otherwise set `use_condition=OFF` and don't build/use the event system.

**Comment by darobs** [comment]
Submitted #267 to address this issue. 

Setting the cmake "enable_event_system" to OFF will disable the event system reporting (which uses the c-shared Condition code), and sets "use_condition" to OFF for the c-shared build.

**Comment by damonbarry** [comment]
Fixed by c04f92a.

**Comment by damonbarry** [comment]
Available in the 2017-08-21 release.
